core/controllers/suggestion.py

```diff
@@ -139,6 +187,58 @@ def put(self, target_id, suggestion_id):
         self.render_json(self.values)
 
 
+class ReviewableSuggestionsHandler(base.BaseHandler):
+    """Provides all suggestions which can be reviewed by the user for a given
+    suggestion type.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_access_admin_page
+    def get(self, target_type, suggestion_type):
+        """Handles GET requests."""
+        try:
+            _require_valid_suggestion_and_target_types(
+                target_type, suggestion_type)
+
+            suggestions = suggestion_services.get_reviewable_suggestions(
+                self.user_id, suggestion_type)
+
+            if target_type == suggestion_models.TARGET_TYPE_EXPLORATION:
+                self.render_json(_get_exploration_suggestion_representable(
+                    suggestions))
+            else:
+                self.render_json({})
+        except Exception as e:
+            raise self.InvalidInputException(e)
+
+
+class UserSubmittedSuggestionsHandler(base.BaseHandler):
+    """Provides all suggestions which can be reviewed by the user for a given
+    suggestion type.
+    """
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_suggest_changes
+    def get(self, target_type, suggestion_type):
+        """Handles GET requests."""
+        try:
+            _require_valid_suggestion_and_target_types(
+                target_type, suggestion_type)
+
+            suggestions = suggestion_services.get_submitted_suggestions(
+                self.user_id, suggestion_type)
+
+            if target_type == suggestion_models.TARGET_TYPE_EXPLORATION:
+                self.render_json(_get_exploration_suggestion_representable(
```

Comment:
While getting suggestions, why do you also need the opportunities? Basically, what's the difference between this handler and the above one?

Reply:
> While getting suggestions, why do you also need the opportunities?

We need information like topic and story details in frontend to represent the suggestion for the reviewers/authors, opportunity dict will provide all such information.

Reply:
> what's the difference between this handler and the above one?

The above one is for reviewers and this one is for users who want to see the list of suggestions they have made.

Reply:
Oh ok... That sounds reasonable.